app/models/publication.rb

```diff
@@ -80,6 +80,7 @@ def self.open_access_statuses
         }
 
   scope :subject_to_open_access_policy, -> { journal_article.published.where('published_on >= ?', Publication::OPEN_ACCESS_POLICY_START) }
+  scope :claimable_by, ->(user) { journal_article.visible.where.not(id: user.authorships.unclaimable.map { |a| a.publication.id }) }
```

Comment:
Unless authorships automatically preloads its publication, then there's an n+1 here. `a.publication.id` will hit the database for every iteration to load the publication out, simply to ask it for its id, which Authorship already knows. `a.publication_id` could accomplish the same thing without loading out another activerecord

Extra macho factor: `user.authorships.unclaimable.map(&:publication_id)` or `user.authorships.unclaimable.pluck(:publication_id)` depending on whether those authorships are loaded or not.